src/openssl_wrap.cpp

```diff
@@ -1250,6 +1292,23 @@ EC_KEY *_EVP_PKEY_get0_EC_KEY(EVP_PKEY *pkey)
     return OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_EVP_PKEY_get0_EC_KEY,pkey);
 }
 
+void _ECDSA_SIG_set0(ECDSA_SIG* sig, SSL_BIGNUM_Ptr r, SSL_BIGNUM_Ptr s) {
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_ECDSA_SIG_set0, sig, r.release(), s.release());
+}
+
+std::vector<uint8_t> _i2d_ECSDA_SIG(const ECDSA_SIG *sig) {
+    OpenSSLGuardedOutputBuffer<unsigned char> outputBuffer;
+    int result = lib::OpenSSLLib::SSL_i2d_ECDSA_SIG(sig, &outputBuffer.get());
```

Comment:
`outputBuffer.get()` is a `nullptr` here, you're passing the address of that `nullptr` in the `outputBuffer` object. How is the write operation that follows not a null pointer dereference?

The documentation says
> i2d_ECDSA_SIG() creates the DER encoding of the ECDSA signature sig and writes the encoded signature to *pp (note: if pp is NULL i2d_ECDSA_SIG() returns the expected length in bytes of the DER encoded signature). i2d_ECDSA_SIG() returns the length of the DER encoded signature (or 0 on error).

The way I see this you'd have to call `SSL_i2d_ECDSA_SIG(sig, nullptr)` and use the length to allocate the output buffer, then call it again. What am I missing?

Reply:
The function expects `unsigned char **pp`, which is a classical pointer output parameter in which memory is reserved for you. This is actually also what happens. I'll add a comment w.r.t. confusing OpenSSL documentation